agent-connection/src/main/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/AgentWebSocketClient.kt

```diff
@@ -0,0 +1,211 @@
+package com.luxoft.blockchainlab.corda.hyperledger.indy
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.luxoft.blockchainlab.hyperledger.indy.utils.SerializationUtils
+import mu.KotlinLogging
+import org.java_websocket.client.WebSocketClient
+import org.java_websocket.handshake.ServerHandshake
+import rx.*
+import java.net.URI
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedDeque
+
+class AgentWebSocketClient(serverUri: URI, private val socketName: String) : WebSocketClient(serverUri) {
+    private val log = KotlinLogging.logger {}
+
+    override fun onOpen(handshakedata: ServerHandshake?) {
+        log.info { "$socketName:AgentConnection opened: $handshakedata" }
+    }
+
+    override fun onClose(code: Int, reason: String?, remote: Boolean) {
+        log.info { "$socketName:AgentConnection closed: $code,$reason,$remote" }
+    }
+
+    /**
+     * inbound messages list
```

Comment:
Please, specify what are the map's arguments

Reply:
done.